src/dumch/parse.cljc

```diff
@@ -247,7 +248,7 @@
     :return (if v1 (ast->clj v1) (tnode 'nil))
     :typecasting (ast->clj v1)
     :const (n/meta-node (tnode :const) (ast->clj v1))
-    :identifier (symbol (str/replace v1 #"!" ""))
+    :identifier (symbol (str/replace (str v1) #"!" ""))
```

Comment:
I think the only source of symbols here is the `flatten-commutative-node` method.  Lets just change it to
```clojure
(defn- flatten-commutative-node [[f :as node]]
  (list* 
    [:identifier (case f :and "and", :or "or", :add "+", :mul "*")]
    (flatten-same-node node)))
```